#2537 Give flags a ring-shadow, to make the white parts visible

mpsijm opened the pull request:
Fixes #2537.

The issue description suggested to add a border, but since this is seen as part of the width/height of the image, this messes up the ratio of the flag (same goes for `padding` btw, so I changed that to `margin`). So, I added a shadow instead.
Note that the flags are now slightly smaller than they used to be because I used round `rem` values for width and height – I hope that's fine, and else we can still increase them :slightly_smiling_face:
Same goes for the hardness of the shadow – `0.3` looks fine to me, but perhaps others have different opinions :smile: 

| Before | After |
|---|---|
| ![image](https://github.com/DOMjudge/domjudge/assets/9739541/dcc14ab0-f1d8-441e-a86d-6b51a10fba96) | ![image](https://github.com/DOMjudge/domjudge/assets/9739541/5ed3b277-f501-4dc1-b286-6d6f47947acb) |
| ![image](https://github.com/DOMjudge/domjudge/assets/9739541/49c0e0c4-6bd5-4322-8577-ad448bac6310) | ![image](https://github.com/DOMjudge/domjudge/assets/9739541/cc1b95ac-223f-4aa7-9ff0-7974f48e73d5) |

vmcj commented:
![image](https://github.com/DOMjudge/domjudge/assets/14887731/46c50508-2a4f-404a-a018-4e838143f694)


It does fix the issue, but it looks strange that its the only item popping out (but probably nothing we can do about that..)

mpsijm commented:
We could make the shadow smaller to lessen the effect, but there's a fine balance there, because if the shadow is too small, it still doesn't fix the issue. We could also use a non-fading `box-shadow` (like [this so-called "ring"](https://tailwindcss.com/docs/ring-width)), will try that tomorrow :slightly_smiling_face:

mpsijm commented:
I changed the fading shadow into a solid `1px` "ring"-shadow with hardness `0.2` (again, this can be changed). What do you think? :smile: 
Also note that this ring is now in the margin, so we may want to slightly increase this margin.

![image](https://github.com/DOMjudge/domjudge/assets/9739541/7b23c08f-7fa7-40a4-93e4-00b7641e05a9)
![image](https://github.com/DOMjudge/domjudge/assets/9739541/70809c1a-3cde-4f05-9644-c8bb0dad6da7)

In the screenshot of Demoweb below, I manually changed the styling rules and replaced one of the UK flags with the Polish flag from the browser's Dev Console.
![image](https://github.com/DOMjudge/domjudge/assets/9739541/2c237cb5-dd67-4393-91f1-e9bac1706495)

vmcj reviewed (approved):
I like it, lets give the rest a day or two to look at it and otherwise I'll merge it.

nickygerritsen commented:
This looks great
